esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -866,19 +869,25 @@ void EsdlServiceImpl::generateTargetURL(IEspContext & context,
 
     StringBuffer name(srvinfo->queryProp("@name"));
 
-    ISmartSocketFactory *conn = connMap.getValue(name);
-    if (!conn)
+    CSmartSocketFactory *sconn = static_cast<CSmartSocketFactory*>(connMap.getValue(name));
```

Comment:
connMap only contains instances of CSmartSocketFactory. however the interface change will make this static cast go away.